sql/core/src/test/resources/sql-tests/inputs/transform.sql

```diff
@@ -183,3 +189,84 @@ SELECT a, b, decode(c, 'UTF-8'), d, e, f, g, h, i, j, k, l FROM (
     NULL DEFINED AS 'NULL'
   FROM t
 ) tmp;
+
+SELECT TRANSFORM (b, a, CAST(c AS STRING) )
+USING 'cat' AS (a, b, c)
+FROM script_trans
+WHERE a <= 4;
+
+SELECT TRANSFORM (1,2,3)
+USING 'cat' AS (a, b, c)
+FROM script_trans
+WHERE a <= 4;
+
+SELECT TRANSFORM(1,2)
+USING 'cat' AS (a INT,b INT)
+FROM script_trans
+LIMIT 1;
+
+SELECT TRANSFORM (
+b AS d5, a,
+CASE
+  WHEN c > 100 THEN 1
+  WHEN c < 100 THEN 2
+ELSE 3 END)
+USING 'cat' AS (a,b,c)
+FROM script_trans
+WHERE a <= 4;
+
+SELECT TRANSFORM (b, a, c + 1 )
+USING 'cat' AS (a,b,c)
+FROM script_trans
+WHERE a <= 4;
+
+SELECT TRANSFORM (*)
+USING 'cat' AS (a,b,c)
+FROM script_trans
+WHERE a <= 4;
+
+SELECT TRANSFORM (b AS d, MAX(a) as max_a, CAST(SUM(c) AS STRING))
```

Comment:
In hive, one can use an aggregate filter, e.g., `SELECT TRANSFORM (b AS d, MAX(a) FILTER (WHER a > 3) as max_a`? Could you add tests for it?

Reply:
> In hive, one can use an aggregate filter, e.g., `SELECT TRANSFORM (b AS d, MAX(a) FILTER (WHER a > 3) as max_a`? Could you add tests for it?

Add a test case.